sklearn/neighbors/dist_metrics.pyx

```diff
@@ -1093,22 +1093,29 @@ cdef class PyFuncDistance(DistanceMetric):
         self.func = func
         self.kwargs = kwargs
 
+    # in cython < 0.26, GIL was required to be acquired during definition of
```

Comment:
I don't understand why we *needed* to acquire it twice previously. Seems strange. But I've not been following the discussions.